### PR TITLE
Investigate using Nudge with Objective C iOS projects #179

### PR DESCRIPTION
**Pull request by bromeostasis**
Closes getlarky/nudge#179

Testing [here](https://github.com/getlarky/nudge/issues/179#issuecomment-418493645)

### CODE REVIEW

**Comment by bromeostasis** [comment]
@ishotjr lmk what you think about committing the `specs-zip` changes without getlarky/nudge#261 completed. If you want, I could release a version 0.1.3 that uses the old git method for `source`!

Otherwise, hopefully this takes you 1/10000th the time of testong :bowtie:  🔎 :octocat:

**Comment by ishotjr** [comment]
@bromeostasis not certain I understand - are you saying, like...technically they should have been separated and this should have been just `:git` and then #261 w/b the `:http`?  If so, then, as much as I love pedantry, there's no need for that in this case I don't think.  ~~I do want to get the main repo private ASAP (and re-do all of our security everywhere as required) though (which I realize is part of #261 / #269) - to clarify,~~ which spec repo/source will we be pointing customers to after this? ~~i.e. can't we make the main repo private now and use the public spec but zip source?  Which is guess is kind of #261 "lite" - why do we even care about making the spec repo private?  Aaand I guess ALL if this is somewhat moot until we have obfuscation since we'll still be exposing all of our source and secrets.~~  Aaaaaaand I guess this should all have been in #261/#269 anyway.  Oh well.

File: Example/Podfile [hunk 1]
```diff
@@ -12,3 +12,14 @@ target 'nudge_Example' do
     pod 'Nimble-Snapshots' , '~> 6.3.0'
   end
 end
+# https://github.com/mxcl/PromiseKit/issues/722#issuecomment-350856514
```
Comment (ishotjr) [inline Example/Podfile]: thanks for preserving my link! 😍

File: Example/Pods/Local Podspecs/nudge.podspec.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "nudge",
-  "version": "0.1.0",
-  "summary": "A short description of nudge.",
-  "description": "Add push and location-based notifications to your app. Add push and location-based notifications to your app. Add push and location-based notifications to your app.",
+  "version": "0.1.5",
+  "summary": "Nudge is a powerful tool that enables mobile banking apps to send push notifications to users.",
+  "description": "Nudge is a powerful tool that enables mobile banking apps to send push notifications to users. Register today!",
```
Comment (ishotjr) [inline Example/Pods/Local Podspecs/nudge.podspec.json]: nice move! 👍 🔤

File: Example/Pods/Local Podspecs/nudge.podspec.json [hunk 3]
```diff
@@ -12,8 +12,7 @@
     "bromeostasis": "evan.snyder92@gmail.com"
   },
   "source": {
-    "git": "https://github.com/getlarky/nudge-ios.git",
-    "tag": "0.1.0"
+    "http": "https://github.com/getlarky/nudge-specs-zip/releases/download/0.1.5/nudge.zip"
```
Comment (ishotjr) [inline Example/Pods/Local Podspecs/nudge.podspec.json]: I'd intended this as a quick solution "for now" - but ... urm ... there's not really any reason not to just keep doing it this way at the moment, right?  Do we need to develop a protocol to ensure it gets updated though now that we're not pointing directly to the source repo?

File: Example/nudge.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -481,6 +481,7 @@
 				ONLY_ACTIVE_ARCH = YES;
 				SDKROOT = iphoneos;
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 3.0;
```
Comment (ishotjr) [inline Example/nudge.xcodeproj/project.pbxproj]: does it matter that this is not the same value as the one I added to the podfile (`3.2`)?  I didn't know this was a new addition - any chance that's why I had the Swift version problems in the first place, which I'm assuming RM didn't when he tested the prior version?

File: Example/Pods/Local Podspecs/nudge.podspec.json (hunk 3 above)
Comment (ishotjr) [reply Example/Pods/Local Podspecs/nudge.podspec.json]: *maybe this comment should have been at https://github.com/getlarky/nudge-ios/pull/4/files#diff-bab6542b515283e26609a3d1f6359fb2R28*

**Comment by ishotjr** [comment]
*just realized: I need to review doc changes too...* :bowtie: 🔎 📜

**Comment by bromeostasis** [comment]
> not certain I understand - are you saying, like...technically they should have been separated and this should have been just :git and then getlarky/nudge#261 w/b the :http? If so, then, as much as I love pedantry, there's no need for that in this case I don't think.

@ishotjr 🐦 , yeah that's what I was saying. I guess I was concerned about closing this with the potentially incomplete/untested getlarky/nudge#261 process, but this link should always work, so we should be good to do this for now and update later.

> which spec repo/source will we be pointing customers to after this?

@ishotjr hrmm, I think let's keep the docs as-is until we either:

a) Get a customer who needs objective c support
b) Finish getlarky/nudge#261

File: Example/Pods/Local Podspecs/nudge.podspec.json (hunk 3 above)
Comment (bromeostasis) [reply Example/Pods/Local Podspecs/nudge.podspec.json]: @ishotjr yeah, I suppose after PR is complete, a new zip is posted, a new release is created, and the podspec is updated with the latest version? Does that sound good?

**Comment by ishotjr** [comment]
>> not certain I understand - are you saying, like...technically they should have been separated and this should have been just :git and then getlarky/nudge#261 w/b the :http? If so, then, as much as I love pedantry, there's no need for that in this case I don't think.
> @ishotjr 🐦 , yeah that's what I was saying. I guess I was concerned about closing this with the potentially incomplete/untested getlarky/nudge#261 process, but this link should always work, so we should be good to do this for now and update later.

@bromeostasis 🐦 

>> which spec repo/source will we be pointing customers to after this?

> @ishotjr hrmm, I think let's keep the docs as-is until we either:

> a) Get a customer who needs objective c support
> b) Finish getlarky/nudge#261 

@bromeostasis :bird:

**Comment by bromeostasis** [comment]
> Under https://github.com/getlarky/nudge-documentation/wiki/nudge-iOS-integration-instructions#adding-the-nudge-pod-to-your-project it strikes me as atypical to see the trailing ;s here - can we remove?



> Under https://github.com/getlarky/nudge-documentation/wiki/nudge-iOS-integration-instructions#integrating-the-nudge-library-with-your-code can we make sample AppDelegate at the bottom of this doc a real link instead of the * thing?

Changes [made](https://github.com/getlarky/nudge-documentation/wiki/nudge-iOS-integration-instructions)!

> I didn't take an ultra-close look at docs for typos etc. vs. just making sure the steps worked for testing - LMK if you'd like me to.

@ishotjr I'm sure we'll be ok 😄 . Lemme know if ready for final 🚀

File: Example/Pods/Local Podspecs/nudge.podspec.json (hunk 3 above)
Comment (ishotjr) [reply Example/Pods/Local Podspecs/nudge.podspec.json]: @bromeostasis coo! And we're confident we'll remember, right? 😅

File: Example/nudge.xcodeproj/project.pbxproj (hunk 4 above)
Comment (ishotjr) [reply Example/nudge.xcodeproj/project.pbxproj]: @bromeostasis haha perf. 🚀

**Comment by ishotjr** [comment]
@bromeostasis awesome - thank you! 🚀

**Comment by bromeostasis** [comment]
> @ishotjr yeah, I suppose after PR is complete, a new zip is posted, a new release is created, and the podspec is updated with the latest version? Does that sound good?

Ok crap I meant after a CR is completed and *before* merge happens, so we don't allow ourselves to close the issue until the update is made. I'll reopen to remind me.

**Comment by ishotjr** [comment]
@bromeostasis like, including this one? Oh. Sorry! 😅

**Comment by bromeostasis** [comment]
> @bromeostasis like, including this one? (I thought we did that already here). Sorry! 😅

@ishotjr oops, I was worried about [this commit](https://github.com/getlarky/nudge-ios/pull/4/commits/9da7c22cce7cd65359d874fe54f407b04c1915e2) which came after the nudge-specs release, so I think we're actually good. My bad!

**Pull request merged**